src/state/opamFormatUpgrade.ml

```diff
@@ -1098,14 +1098,28 @@ let as_necessary requested_lock global_lock root config =
     |> List.partition (fun (v,_) ->
         OpamVersion.compare v latest_hard_upgrade <= 0)
   in
+  let erase_plugin_links root =
+    let root = OpamFilename.Dir.to_string (OpamPath.plugins_bin root) in
+    let files = Sys.readdir root in
+    let erase_link file =
+      (* Sys.file_exists will be false for a dangling symlink *)
```

Comment:
This surprised me, @altgr, looking at `OpamSystem.files_with_links`, I think it doesn't return dangling symlinks. Is that a bug? If it is, then it can be fixed and used here instead.